### ISSUE TITLE
Sending Authorization: Bearer <token> header

### ISSUE THREAD

**Issue by chornyi** [springfox/springfox#1123]
I would like to enter `"Bearer <token>"` in the API Key field and have a header "Authorization: Bearer <token>" to be sent to the server. According to Swagger UI documentation this should be possible: https://github.com/swagger-api/swagger-ui#custom-header-parameters---for-basic-auth-etc. Can Springfox be configured to do this? I tried configuring security scheme like this:

    private ApiKey apiKey() {
        return new ApiKey("apikey", "Authorization", "header");
    }

But this still results in header `api_key: Bearer%20<token>` being sent to the server. In other words, I am unable to change the name of the header. Also notice that the API key value is URL encoded, which might be another problem.

I reviewed #943 and https://github.com/swagger-api/swagger-ui/issues/301 but it is still not clear if my use case is supported.

I am using version `2.3.1`.

**Comment by chornyi** [comment]
I noticed this might be a duplicate of https://github.com/springfox/springfox/issues/1087. Please feel free to close if this is the same issue.

**Comment by dilipkrish** [comment]
@chornyi I think this is a duplicate. I'll close this for now. I could certainly use some help fixing these issues including #1126.

**Comment by dtritus** [comment]
I want to ask you to reopen it, since issue with API key value being URL encoded is not fixed.
It happens [here ](https://github.com/springfox/springfox/blob/3ce628bf00a2b341709706acf9edc2f620af3448/springfox-swagger-ui/src/web/js/springfox.js#L52) and I don't understand what is the reason to encode API key value.

**Comment by elennick** [comment]
:+1: having the same issue trying to set a basic auth header in 2.4

**Comment by dilipkrish** [comment]
Waiting on this pr https://github.com/springfox/springfox/pull/1218 to get merged in. I still have an outstanding question before I merge it in

**Comment by mika75** [comment]
is there any away to set a header ? : 
for exemple : Authorization Bearer ....

Iam using  2.6.1

**Comment by yeliseevalex** [comment]
to change the name of the header you can use 
@Bean
SecurityConfiguration security() {
		return new SecurityConfiguration(null, null, null, null, "Bearer access_token",    ApiKeyVehicle.HEADER, "Authorization", ",");
}

**Comment by dilipkrish** [comment]
Thanks @yeliseevalex

**Comment by mika75** [comment]
Thanks @yeliseevalex

**Comment by tareqmy** [comment]
@yeliseevalex can you please tell me, this 'access_token' is it hardcoded from server? I am storing a variable named 'jwt_token' in localStorage when a user logs in. Can this 'jwt_token' be used from the swagger-ui when this authorization header is sent with a http request? Or how to set this 'access_token' when a user authenticates? 

Even having a text field in swagger-ui where i can manually put the server returned token?

excuse me, if my questions does not make much sense. I could not understand from any of documents or searching online.

thanks.

**Comment by rdemorais** [comment]
@tareqmy same problem here...

**Comment by dilipkrish** [comment]
@tareqmy It makes total sense. The swagger-ui that is bundled with springfox is a 80% solution. For more advanced/custom use cases, it doesnt make sense to use it. A better option in that case would be to import the swagger-ui distributable and customize it to your needs. A good example of doing that is the jhipster project which handles the JWT token just like how you want it.

/c @rdemorais

**Comment by mika75** [comment]
It could be nice that Value field in "Available authorizations" Popup could used for set Header
Bearer + token

**Comment by dilipkrish** [comment]
2.7 and 2.8 have different versions of swagger you @prabhatCH. Also this is a really old issue. Please consider creating a new issue with more details, so I can help

**Comment by losty-4** [comment]
@yeliseevalex Thanks
@mika75 trying using swagger 2.5
